ext/thread_test/puppetserver.conf

```diff
@@ -0,0 +1,200 @@
+    # any client.  For better security, you might wish to set this to 'localhost'.
+    host: localhost
+    port: 8140
+}
+
+# configure the mount points for the web apps
+web-router-service: {
+    # These two should not be modified because the Puppet 4.x agent expects
+    # them to be mounted at these specific paths
+    "puppetlabs.services.ca.certificate-authority-service/certificate-authority-service": "/puppet-ca"
+    "puppetlabs.services.master.master-service/master-service": "/puppet"
+
+    # This controls the mount point for the puppet admin API.
+    "puppetlabs.services.puppet-admin.puppet-admin-service/puppet-admin-service": "/puppet-admin-api"
+
+    # 3.x agent compatibility routes. The legacy routes are only mounted if the
+    # puppetlabs.services.legacy-routes.legacy-routes-service/legacy-routes-service
+    # service is in the bootstrap.cfg
+    "puppetlabs.services.legacy-routes.legacy-routes-service/legacy-routes-service": ""
+
+    # This controls the mount point for the status API
+    "puppetlabs.trapperkeeper.services.status.status-service/status-service": "/status"
+
+    # This controls the mount point for the metrics API
+    "puppetlabs.trapperkeeper.services.metrics.metrics-service/metrics-webservice": "/metrics"
+}
+
+# configuration for the JRuby interpreters
+jruby-puppet: {
+    # Where the puppet-agent dependency places puppet, facter, etc...
+    # Puppet server expects to load Puppet from this location
+    ruby-load-path: [./ruby/puppet/lib, ./ruby/facter/lib, ./ruby/hiera/lib, ./ruby/resource_api/lib]
+
+    # This setting determines where JRuby will install gems.  It is used for loading gems,
+    # and also by the `puppetserver gem` command line tool.
+    gem-home: ${HOME}"/.puppetlabs/opt/server/data/puppetserver/jruby-gems"
+
+    # This setting defines the complete "GEM_PATH" for jruby.  If set, it should include
+    # the gem-home directory as well as any other directories that gems can be loaded
+    # from (including the vendored gems directory for gems that ship with puppetserver)
+    gem-path: [${jruby-puppet.gem-home}, ${HOME}"/.puppetlabs/opt/server/data/puppetserver/vendored-jruby-gems"]
+
+    # (optional) path to puppet conf dir; if not specified, will use the puppet default
+    master-conf-dir: ${HOME}"/.puppetlabs/etc/puppet"
+
+    # (optional) path to puppet code dir; if not specified, will use the puppet default
+    master-code-dir: "./ext/thread_test"
+
+    # (optional) path to puppet var dir; if not specified, will use the puppet default
+    master-var-dir: ${HOME}"/.puppetlabs/opt/puppet/cache"
+
+    # (optional) path to puppet run dir; if not specified, will use the puppet default
+    master-run-dir: ${HOME}"/.puppetlabs/var/run"
+
+    # (optional) path to puppet log dir; if not specified, will use the puppet default
+    master-log-dir: ${HOME}"/.puppetlabs/var/log"
+
+    # (optional) maximum number of JRuby instances to allow
+    max-active-instances: 1
```

Comment:
With the ReferencePool, this setting is used to control the number of threads that can go through the instance at once. Once we have integrated that, we probably want to set it higher to properly test the multithreaded behavior.

Related tangential question: should we double up on the usage of this setting, or create new ones specific to the multithreaded environment?

Reply:
Honestly I'm not sure

Reply:
We can leave the setting the same for now and update it if it seems bad or confusing once we show it to other people. I think it's okay if the API is a little fluid at the start.